### PR TITLE
Make iOS project path configurable by parameter in run-ios cli command

### PR DESCRIPTION
**Pull request by mroswald**
In projects where you have multiple apps or a different structure (ios is not the directory containing the xcode project) it would be helpful to have the option to configure the directory containing the xcode project when doing "react-native run-ios"

As my PR does not change and does not affect UI it's hard to show a video ;-) Steps to reproduce are:

1. checkout facebook/react-native
2. run 'react-native run-ios' (it will fail)
3. npm link react-native (with my PR)
4. re-run 'react-native run-ios --ios-project-path Examples/Movies/

Now the simulator should open the Movies example app

### CODE REVIEW

File: local-cli/runIOS/runIOS.js [hunk 1]
```diff
@@ -27,15 +27,24 @@ function runIOS(argv, config) {
...
+  const args = parseCommandLine([
+    {
+      command: 'simulator',
+      description: 'Explicitly set simulator to use',
+      type: 'string',
+      required: false,
+      default: 'iPhone 6',
+    },
+    {
+      command: 'ios-project-path',
```
Comment (grabbou) [inline local-cli/runIOS/runIOS.js]: I think it should be `project-dir` instead since it's not really a path? Apart from that, looks good.

**Comment by grabbou** [comment]
@mkonicek @frantic @Kureev that looks good to me. We are going to probably improve that in the future anyway, but it will finally allow me to run my nested apps w/o issues.

File: local-cli/runIOS/runIOS.js (hunk 1 above)
Comment (mroswald) [inline local-cli/runIOS/runIOS.js]: Shall I change that?

File: local-cli/runIOS/runIOS.js [hunk 2]
```diff
@@ -27,15 +27,24 @@ function runIOS(argv, config) {
...
 function _runIOS(argv, config, resolve, reject) {
-  const args = parseCommandLine([{
-    command: 'simulator',
-    description: 'Explicitly set simulator to use',
-    type: 'string',
-    required: false,
-    default: 'iPhone 6',
-  }], argv);
+  const args = parseCommandLine([
+    {
```
Comment (Kureev) [inline local-cli/runIOS/runIOS.js]: can we also stick to the current style rules? (e.g. `{` on the same line)

**Comment by Kureev** [comment]
I have no problems with having an additional parameter, but the final word would be on @mkonicek

File: local-cli/runIOS/runIOS.js [hunk 3]
```diff
@@ -27,15 +27,24 @@ function runIOS(argv, config) {
...
+      command: 'simulator',
+      description: 'Explicitly set simulator to use',
```
Comment (mkonicek) [inline local-cli/runIOS/runIOS.js]: Please stick to the conventions used across the project, for example "Simulator to use, for example 'iPhone 6'."

Also, is this only a simulator, or also possibly a physical device?

See e.g. https://github.com/facebook/react-native/blob/3a92f2017fc2e51c2a915f0e535397cb3937d412/local-cli/generate/generate.js

File: local-cli/runIOS/runIOS.js (hunk 1 above)
Comment (mkonicek) [inline local-cli/runIOS/runIOS.js]: I'm fine with `ios-project-path`, it makes it clear it's the path where the XCode project lives.

Regarding the "path" vs "dir" terminology - if you look at other places in the CLI, we mostly use "path" so let's use that.

We should also add this for Android, can you add that to this PR please or send a separate PR?
`react-native run-android --android-project-path Examples/Movies/`

File: local-cli/runIOS/runIOS.js [hunk 4]
```diff
@@ -27,15 +27,24 @@ function runIOS(argv, config) {
...
+      type: 'string',
+      required: false,
+      default: 'iPhone 6',
+    },
+    {
+      command: 'ios-project-path',
+      description: 'Set XCode project path',
```
Comment (mkonicek) [inline local-cli/runIOS/runIOS.js]: nit: "Path relative to project root where the XCode project (.xcodeproj) lives. The default is 'ios'."

**Comment by mkonicek** [comment]
Just nits. Thanks for doing this!

File: local-cli/runIOS/runIOS.js (hunk 1 above)
Comment (frantic) [inline local-cli/runIOS/runIOS.js]: @mkonicek I'd disagree with `ios-project-path`, because this is not something that wouldn't make sense on Android (or other platform) one day. Also the command is `run-ios` which already sets the context for `ios`. `simulator` param to this command is also iOS-specific but it doesn't have the prefix.

Let's stick to `project-path`?

File: local-cli/runIOS/runIOS.js (hunk 3 above)
Comment (mroswald) [inline local-cli/runIOS/runIOS.js]: I didn't change anything here besides spaces ;-) But I can change the text in this PR anyway

File: local-cli/runIOS/runIOS.js (hunk 1 above)
Comment (mroswald) [inline local-cli/runIOS/runIOS.js]: I'd suggest to discuss the change for android in another PR. There is already a parameter "root", which seems to do the same job. Maybe we just rename it or add an alias (for backwards compability). Also we have no working sample app for android which is runnable via run-android (needs a android/gradlew file?).

**Comment by grabbou** [comment]
Thanks for the updates! I know these styling issues might be pretty annoying to fix. Is `eslint` not reporting any warnings?

https://github.com/facebook/react-native/blob/master/.eslintrc

**Comment by mroswald** [comment]
No, it doesn't. Maybe you want to stick to rules like from airbnb? Or provide another "standard" set of rules in a separate project?

**Comment by mroswald** [comment]
@grabbou is there anything left I have to do or update?

**Comment by grabbou** [comment]
That looks good to me. Going to ship it (hope @mkonicek doesn't mind that).

**Comment by grabbou** [comment]
@facebook-github-bot shipit

**Comment by grabbou** [comment]
Regarding the linting issues, would you mind opening a thread on our Facebook group -> https://www.facebook.com/groups/react.native.community/ and cc'ing me & Martin?

**Comment by mroswald** [comment]
sure :-)

**Comment by mkonicek** [comment]
I don't know about lint but saw the post and agree it would be impactful if someone could improve it.

**Comment by mkonicek** [comment]
Failed with a merge conflict. Can you rebase please?

**Comment by mkonicek** [comment]
Rebasing will rerun the tests.

**Comment by mroswald** [comment]
What's missing?

**Comment by mkonicek** [comment]
@mroswald You just need to ping us sometimes. Didn't get any notification that you rebased.

@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by mroswald** [comment]
@mkonicek thanks! I will ping you in the future, stay tuned ;-)

**Comment by mkonicek** [comment]
@mroswald I mean ping someone who has context on the code you're changing (see git blame), not me :)
